rosdep/python.yaml

```diff
@@ -4047,6 +4047,24 @@ python3-pep8:
 python3-pkg-resources:
   debian: [python3-pkg-resources]
   ubuntu: [python3-pkg-resources]
+python3-pyparsing:
+  arch: [python-pyparsing]
+  debian: [python3-pyparsing]
+  fedora: [python3-pyparsing]
+  gentoo: [dev-python/pyparsing]
+  ubuntu: [python3-pyparsing]
+python3-pytest:
+  arch: [python-pytest]
+python3-pytest-pip:
```

Comment:
Which package(s) is/are going to use this key?

Reply:
I have no explicit use case for this key, I was attempting to follow observed convention that keys using pip were separate from the regular rosdep key.

It's possible that a platform or OS variable could be used with Package format 3 dependency conditions to select the pip dependency on platforms where it is not available from the default provider.

Reply:
I would suggest not to add rosdep keys which are not needed / used by any package but wait with such an addition until that is the case.

Reply:
Does it matter that dropping this would mean that no rosdep dependency key for ubuntu xenial (the supported platform) satisfies the test dependencies for ROS 2?

Reply:
Before you said the key isn't used anywhere. Under that assumption I recommended to not add it (since it would not be used anywhere).

Can you please clarify which package currently uses the key?

Reply:
> Can you please clarify which package currently uses the key (or is going to be updated to use this key)?

No packages currently use the key. `ament_cmake_pytest` depends on https://github.com/ament/ament_cmake/blob/master/ament_cmake_pytest/package.xml#L14 `python3-pytest` but as you previously mentioned, apparently the version in apt isn't sufficient. Since tests don't abort buildfarm builds, this did not block the release.

There's also a significant number of packages which have `test_requires=['pytest']` in the setup.py but no entry for this in the package.xml.

I don't have much stake in whether the solution is using the pip provider in the "generic" key or switching to use the pip key knowing that our supported platform can't be supported without it. The only way I can think to have cake and eat it too is by trying to use the new dependency conditions to apply platform specific dependencies.

Reply:
> The only way I can think to have cake and eat it too is by trying to use the new dependency conditions to apply platform specific dependencies.

Can you please make a concrete example. Currently I can't follow you how this would look like.

Reply:
```
<depend condition="$OS_CODENAME == bionic">python3-pytest</depend>
<depend condition="$OS_CODENAME != bionic">python3-pytest-pip</depend>
```

Reply:
That would imply that the package can't be released with bloom into distributions other than `bionic`.

Reply:
Right, I forgot that bloom hard-stops on pip dependencies.

There was a proposed change to allow them to be skipped instead. https://github.com/ros-infrastructure/bloom/pull/412

Reply:
Skipping them breaks the expectation that all dependencies are resolved when installing a debian package.

Reply:
> Skipping them breaks the expectation that all dependencies are resolved when installing a debian package.

It isn't my aim to revisit that PR discussion here. I had to remind myself of bloom's behavior and cross-linked that PR as it was part of that investigation.

If this key isn't useful. I'll remove it.